### PR TITLE
feat: ORA BFF Page Data

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Adds top-level info for ORA page data (student/workflow specific data we will need to render any of the views in our MFE).

JIRA: [AU-1382](https://2u-internal.atlassian.net/browse/AU-1382)

**What changed?**

- Added 2 new XBlock handlers:
    - `get_block_learner_submission_data` - Our submission view in the MFE
    - `get_block_learner_assessment_data` - Our assessment view in the MFE
- Implemented serializers for top-level data.
- **NOTE** that at this time I have specifically left the `submission` field unimplemented. This is a large task (since we must gather data from different contexts for each step of the ORA workflow) that should be split up into a few tickets.
- Added beginning of a Grades API and extended the Peer Assessments API to get required data for our serializers.
- Added new operating mode for `get_submission_to_assess` from Peer API. Adding `peek=True` will allow you to check for peers to assess without beginning the assessment workflow. This helps us determine if we are currently blocked waiting for assessments, though in a sub-optimal way since this fetches a lot of data we won't end up using. Thought it was still a better approach than duplicating a lot of code to handle this slightly different use-case.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

- Manual test of data via Postman (or similar API tool) at different parts of the workflow.
- **No unit tests, currently.**, has been ticketed separately in [AU-1464](https://2u-internal.atlassian.net/browse/AU-1464)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Review by jansenk — COMMENTED** [review]
Thankfully it doesn't look like there's actually much duplication here, but there is a fair bit of adjacency / overlap. I suggest you merge this first and then I can rebase my changes on top of / into yours

File: openassessment/xblock/apis/grades_api.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            "points_earned": (Int) awarded points
+            "points_possible": (Int) max possible points
+        }
+        """
+        submission_uuid = self._get_submission_uuid()
+        assessment = self_api.get_assessment(submission_uuid)
+        return {
+            "points_earned": assessment["points_earned"],
+            "points_possible": assessment["points_possible"],
+        }
```
Comment (jansenk) [inline openassessment/xblock/apis/grades_api.py:33]:
```suggestion
        assessment = self_api.get_assessment(submission_uuid)
        if assessment is not None:
            return {
                "points_earned": assessment["points_earned"],
                "points_possible": assessment["points_possible"],
            }
        return None
```

Comment (jansenk) [reply openassessment/xblock/apis/grades_api.py:33]: Same suggestion for peer

File: openassessment/xblock/ui_mixins/mfe/ora_config_serializer.py [hunk 2]
```diff
@@ -217,9 +217,7 @@ class OraBlockInfoSerializer(Serializer):
 
     def get_baseAssetUrl(self, block):
         # pylint: disable=protected-access
-        return block.get_base_url_path_for_course_assets(
```
Comment (jansenk) [inline openassessment/xblock/ui_mixins/mfe/ora_config_serializer.py:220]: Why the filename change for this file?

**Review by jansenk — APPROVED** [review]
looks good. a few comments

File: openassessment/xblock/ui_mixins/mfe/ora_config_serializer.py (hunk 2 above)
Comment (nsprenkle) [reply openassessment/xblock/ui_mixins/mfe/ora_config_serializer.py:220]: I've been splitting the serializers by the response shape they are returning for. This is helping (me at least) keep things separated, since otherwise this would be a 1k+ line file by the time we're done with it which makes finding things harder.

File: openassessment/xblock/apis/grades_api.py (hunk 1 above)
Comment (nsprenkle) [reply openassessment/xblock/apis/grades_api.py]: Good catch, TY
